contracts/oracle/CollateralizationOracle.sol

```diff
@@ -0,0 +1,316 @@
+    ///         from certain deposits could be problematic.
+    /// @param _deposit : the deposit to exclude or re-enable.
+    /// @param _excluded : the new exclusion flag for this deposit.
+    function setDepositExclusion(address _deposit, bool _excluded) external onlyGuardianOrGovernor {
+        excludedDeposits[_deposit] = _excluded;
+    }
+
+    /// @notice Add a PCVDeposit to the list of deposits inspected by the
+    ///         collateralization ratio oracle.
+    ///         note : this function reverts if the deposit is already in the list.
+    ///         note : this function reverts if the deposit's token has no oracle.
+    /// @param _deposit : the PCVDeposit to add to the list.
+    function addDeposit(address _deposit) external onlyGovernor {
+        // if the PCVDeposit is already listed, revert.
+        require(depositToToken[_deposit] == address(0), "CollateralizationOracle: deposit duplicate");
+
+        // get the token in which the deposit reports its token
+        address _token = IPCVDepositV2(_deposit).balanceReportedIn();
+
+        // revert if there is no oracle of this deposit's token
+        require(tokenToOracle[_token] != address(0), "CollateralizationOracle: no oracle");
+
+        // update maps & arrays for faster access
+        depositToToken[_deposit] = _token;
+        tokenToDeposits[_token].add(_deposit);
+        tokensInPcv.add(_token);
+
+        // emit event
+        emit DepositAdd(msg.sender, _deposit, _token);
+    }
+
+    /// @notice Remove a PCVDeposit from the list of deposits inspected by
+    ///         the collateralization ratio oracle.
+    ///         note : this function reverts if the input deposit is not found.
+    /// @param _deposit : the PCVDeposit address to remove from the list.
+    function removeDeposit(address _deposit) external onlyGovernor {
+        // get the token in which the deposit reports its token
+        address _token = depositToToken[_deposit];
+
+        // revert if the deposit is not found
+        require(_token != address(0), "CollateralizationOracle: deposit not found");
+
+        // update maps & arrays for faster access
+        // deposits array for the deposit's token
+        depositToToken[_deposit] = address(0);
+        tokenToDeposits[_token].remove(_deposit);
+
+        // if it was the last deposit to have this token, remove this token from
+        // the arrays also
+        if (tokenToDeposits[_token].length() == 0) {
+          tokensInPcv.remove(_token);
+        }
+
+        // emit event
+        emit DepositRemove(msg.sender, _deposit);
+    }
+
+    /// @notice Swap a PCVDeposit with a new one, for instance when a new version
+    ///         of a deposit (holding the same token) is deployed.
+    /// @param _oldDeposit : the PCVDeposit to remove from the list.
+    /// @param _newDeposit : the PCVDeposit to add to the list.
+    function swapDeposit(address _oldDeposit, address _newDeposit) external onlyGovernor {
```

Comment:
Thoughts on just calling add then remove? It takes a bit more gas but may reduce complexity. This function is pretty simple so I don't feel strongly about it.

Reply:
best code is no code 👍

Reply:
Now that the logic is remove + add I don't think we need to enforce the tokens are the same as they are effectively distinct operations